### PR TITLE
expose HDFS file system stats via Executor metrics

### PR DESCRIPTION
**Pull request by anfeng**
This PR exposes HDFS stats via Executor metrics.

We support 2 HDFS schemes:
- hdfs ... Hadoop distributed file system
- file ... Hadoop local file system

For each scheme, we have the following gauges:
- Number of bytes written
- Number of read operations
- Number of large read operations
- Number of write operations

### CODE REVIEW

**Comment by rxin** [comment]
Jenkins, test this please.

**Comment by rxin** [comment]
Jenkins, test this please.

Jenkins, add to whitelist.

File: core/src/main/scala/spark/executor/ExecutorSource.scala [hunk 1]
```diff
@@ -2,9 +2,24 @@ package spark.executor
...
+
+import scala.collection.JavaConversions._
+
 import spark.metrics.source.Source
 
 class ExecutorSource(val executor: Executor) extends Source {
+  private def fileStats(scheme: String) : Option[FileSystem.Statistics] =
+    FileSystem.getAllStatistics().filter(s => s.getScheme.equals(scheme)).headOption
+
+  private def registerFileSystemStat[T](scheme: String, name: String, f: FileSystem.Statistics => T, defaultValue: T) = {
```
Comment (pwendell) [inline core/src/main/scala/spark/executor/ExecutorSource.scala]: The line limit is 100 character, so this should be wrapped after the `(`

**Comment by rxin** [comment]
Jenkins, test this again .. test this please ...

**Comment by pwendell** [comment]
Hey @anfeng - the code looks good here. However, I just tried to run this both locally and on a cluster and in both cases it didn't work. It looks like FileSystem.getAllStatistics() is just returning an empty list.

Have you tested this at all? Either locally or on a cluster? It doesn't seem to work for me. Is this supported by all HDFS versions?

**Comment by pwendell** [comment]
Ah actually I just realized the tests I was running only read/write shuffle
data... Not Hdfs! Let me run this again tomorrow and if it looks okay I can
merge it.

---

sent from my phone

**Pull request merged**